clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/AdaptiveThrottlerSignalService.java

```diff
@@ -0,0 +1,121 @@
+package com.linkedin.davinci.kafka.consumer;
+
+import com.linkedin.alpini.base.concurrency.Executors;
+import com.linkedin.alpini.base.concurrency.ScheduledExecutorService;
+import com.linkedin.davinci.stats.ingestion.heartbeat.AggregatedHeartbeatLagEntry;
+import com.linkedin.davinci.stats.ingestion.heartbeat.HeartbeatMonitoringService;
+import com.linkedin.venice.service.AbstractVeniceService;
+import io.tehuti.Metric;
+import io.tehuti.metrics.MetricsRepository;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+/**
+ * This class contains service to periodically refresh all the signals for throttlers and update all registered throttler
+ * based on new signal values.
+ */
+public class AdaptiveThrottlerSignalService extends AbstractVeniceService {
+  private static final Logger LOGGER = LogManager.getLogger(AdaptiveThrottlerSignalService.class);
+  public static final double READ_LATENCY_P95_LIMIT = 2.0f;
```

Comment:
Why did we choose p95 for read latency ? also should we consider SLA number, eg p99 > 10ms?

Reply:
Discussed offline, we will update to p99 with a configurable target